src/diffusers/training_utils.py

```diff
@@ -251,6 +251,32 @@ def state_dict(self) -> dict:
             "collected_params": self.collected_params,
         }
 
+    def store(self, parameters: Iterable[torch.nn.Parameter]) -> None:
+        r"""
+        Args:
+        Save the current parameters for restoring later.
+            parameters: Iterable of `torch.nn.Parameter`; the parameters to be
+                temporarily stored.
+        """
+        parameters = list(parameters)
+        self.collected_params = [param.detach().cpu().clone() for param in parameters]
```

Comment:
can we just have the caller do the `detach.cpu.clone` and save collected_params in a variable themselves? This sort of adhoc storage just between two methods is a bit of an OO antipattern imo

Reply:
Another thing is I think the initial `list()` is also unnecessary. If something is an Iterable, I think we should be able to directly iterate over it?

Reply:
Yes, that is true. The detach code was after, hence.